### ISSUE TITLE
Feature Request: Migrate to androidx

### ISSUE THREAD

**Issue by bangarharshit** [facebook/facebook-android-sdk#591]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing feature requests on [GitHub issues](https://github.com/facebook/facebook-android-sdk/issues)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've given my issue the title: `Feature Request: [name of my feature request]`

## Goals

Androidx replaces the android support libraries [official doc](https://developer.android.com/jetpack/androidx/migrate) and facebook sdk is still using the support libraries. Migrate to androidx.

**Comment by gildor** [comment]
Facebook is the only library in our app that still uses the old support library so it prevents us from disabling Jetifier which potentially increase compilation time required to process dependencies. 
I understand, there is backward compatibility, but maybe facebook-sdk should announce at least some point when SDK will be migrated to AndroidX, for example for next major release (6.x?)

**Comment by androideveloper** [comment]
The same here. Waiting for this lib update to drop Jetifier.

**Comment by myflashlab** [comment]
fir anyone interested. we were able to fix the facebook jar files using the [jetifier-standalone](https://developer.android.com/studio/command-line/jetifier). what it does is to parse the jar files and replace AndroidSupport libs with Androidx.

**Comment by gildor** [comment]
Yes, user jetifier for the library will solve the problem, but it requires everyone to publish it to own repository for every FB SDK version

One more solution is to have 2 versions, legacy, and AndroidX (this is what dagger-android does now)

**Comment by TWiStErRob** [comment]
The repo owners (@facebook) could run jetifier standalone and deploy the artifact the usual way, if they don't want to migrate the sources (yet?). I like the idea of two versions, so that everyone can pick the right one. No code changes should be required, just a bit of build setup.

**Comment by sunyal** [comment]
Added a PR to address this https://github.com/facebook/facebook-android-sdk/pull/689

**Comment by MaratGilyazov** [comment]
When will FB migrate to AndroidX? Jetifier solution is not enough, it looks like temp workaround and breaks our workflow! I can't use FB with recent versions of Admob and Facebook, please, migrate to AndroidX, this issue affects my business.

**Comment by gildor** [comment]
For now, we use own fork based on #689 published to our internal repo. Plan to use this version of FB until AndroidX migration is done on upstream

**Comment by R4md4c** [comment]
Just an FYI for people who're still waiting for FB's migration to AndroidX, you can use the suggestion from this comment https://github.com/facebook/facebook-android-sdk/pull/689#issuecomment-607241561, by getting the aars from the maven repository and execute the jetifier command on them, then including the jetified output back as local aars to your project. 

This approach is good for new projects to avoid enabling jeitifier and rising the build time overhead that it brings.

**Comment by TWiStErRob** [comment]
Another workaround without having to commit binary files and doing it manually is to let Jetifier do it with a whitelist for Facebook only (didn't try it with this lib):
```properties
android.useAndroidX=true
android.enableJetifier=true
android.jetifier.blacklist=^(?!.*[\\\\/]com\\.facebook\\.android[\\\\/]).*$
```
See https://github.com/bumptech/glide/issues/3080#issuecomment-495430590 for original idea.

**Comment by mochadwi** [comment]
@TWiStErRob as you pointed out in glide does this `android.jetifier.blacklist=^(?!.*[\\\\/]com\\.facebook\\.android[\\\\/]).*$` regex able to support multi dependencies, such as both glide & facebook included in the lines

**Comment by TWiStErRob** [comment]
Still valid, doesn't require community attention, it requires owner/maintainer attention... 😉 (as said in https://github.com/facebook/facebook-android-sdk/pull/689#issuecomment-590577227)

**Comment by sunyal** [comment]
The SDK has migrated to AndroidX, this is no longer an issue with the latest release

**Comment by TWiStErRob** [comment]
Uuuh, ok, assumed that this issue would be closed along with the 7.0.0 release... 🤦‍♂️ 
https://github.com/facebook/facebook-android-sdk/blob/master/CHANGELOG.md#700---2020-05-05

**Comment by BugsBunnyBR** [comment]
Running [this](https://github.com/dipien/bye-bye-jetifier) tool, It says that the core package depends on bolts-applinks and that depends on android.support...
```
 +---com.facebook.android:facebook-core:8.1.0
                              +---com.parse.bolts:bolts-android:1.4.0
                                   +---com.parse.bolts:bolts-applinks:1.4.0

```
 bolts/MeasurementEvent.class -> android.support.v4.content.LocalBroadcastManager
 
 So, can any facebook dev take a look?

**Comment by androideveloper** [comment]
I also noticed the same using byebyejetifier, didn't check latest version though (9.0+) @BugsBunnyBR

**Comment by lucas34** [comment]
Looks like bolts was removed in v9 release https://github.com/facebook/facebook-android-sdk/commit/8d2755e988f759f0919cd34ce20fd1a5cc3a2626

Can someone confirm ? There is no release note.

**Comment by anthonymonori** [comment]
While there are still no changelog available for 9.0.0, as @lucas34 linked 8d2755e I did verify with [`byebyejetifier`](https://github.com/dipien/bye-bye-jetifier) in a sample project, and it seems to be all gone now (since bolts was the last bottleneck).

ps. if you run [`byebyejetifier`](https://github.com/dipien/bye-bye-jetifier) or other tools to verify it, you may get into a **false positive** since [there is Java reflection api trying to check if the consumer uses an old support library class or not](https://github.com/facebook/facebook-android-sdk/blob/sdk-version-9.0.0/facebook-core/src/main/java/com/facebook/appevents/codeless/internal/ViewHierarchy.java#L251), but otherwise no other references or usages of old Android Support library 🎉

**Comment by linmx0130** [comment]
Close this issue since Android SDK already migrated to androidx.
